pkg/autoscaler/autoscaler.go

```diff
@@ -240,6 +238,17 @@ func (a *Autoscaler) autoscaleFn(api userconfig.Resource) (func() error, error)
 			} else if downscaleStabilizationFloor != nil && request < *downscaleStabilizationFloor {
 				request = *downscaleStabilizationFloor
 			}
+
+			// awaken state: was scaled from zero
+			// This needs to be protected by a Mutex because an Awaken call will also modify it
+			a.Lock()
+			lastAwakenTimestamp := a.lastAwakenTimestamp[api.Name]
+
+			// Make sure we don't scale below zero if API was recently awaken
+			if time.Since(lastAwakenTimestamp) < autoscalingSpec.DownscaleStabilizationPeriod {
+				request = libmath.MaxInt32(request, 1)
+			}
+			a.Unlock()
```

Comment:
Do you know why this is necessary, i.e. that the lines of code above which consider `DownscaleStabilizationPeriod` are not sufficient?

Reply:
it is not sufficient. I have checked it with the unit tests